src/plugins/analysis/qemu_exec/test/test_plugin_qemu_exec.py

```diff
@@ -210,120 +236,212 @@ def test_process_object__included_binary(self):
-        result = qemu_exec.parse_docker_output_options(self.docker_test_output)
-        assert len(result) == 2
-        assert all(option in result for option in [qemu_exec.EMPTY, '--version'])
-
-    def test_parse_docker_output_options__invalid(self):
-        result = qemu_exec.parse_docker_output_options('')
-        assert result == {}
 
-    def test_parse_docker_output_strace__valid(self):
-        result = qemu_exec.parse_docker_output_strace(self.docker_test_output)
-        assert 'strace' in result
-        result['strace'] = decompress(result['strace']).decode()
-        assert result == {'strace': '38 uname(0x7ffffb38) = 0\n38 brk(NULL) = 0x004a2000\n38 brk(0x004a2cc8) = 0x004a2cc8'}
+def test_get_docker_output__static():
+    result = qemu_exec.get_docker_output('mips', '/test_mips_static', TEST_DATA_DIR)
+    _check_result(result)
 
-    def test_parse_docker_output_strace__invalid(self):
-        result = qemu_exec.parse_docker_output_strace('')
-        assert result == {}
 
-    def test_parse_docker_output__valid(self):
-        result = qemu_exec.parse_docker_output(self.docker_test_output)
-        assert len(result) == 3
-        assert all(k in result for k in [qemu_exec.EMPTY, '--version', 'strace'])
+def test_get_docker_output__dynamic():
+    result = qemu_exec.get_docker_output('mips', '/usr/bin/test_mips', TEST_DATA_DIR)
+    _check_result(result)
+
+
+def test_get_docker_output__arm():
+    result = qemu_exec.get_docker_output('arm', '/test_arm_static', TEST_DATA_DIR_3)
+    _check_result(result)
+
+
+def test_get_docker_output__ppc():
+    result = qemu_exec.get_docker_output('ppc', '/test_ppc_static', TEST_DATA_DIR_3)
+    _check_result(result)
+
+
+def _check_result(result):
+    for parameter in CLI_PARAMETERS:
+        assert parameter in result
+        assert 'error' not in result[parameter]
+        assert b'Hello World' in b64decode(result[parameter]['stdout'])
+    assert 'strace' in result
 
-    def test_parse_docker_output__invalid(self):
-        result = qemu_exec.parse_docker_output('')
-        assert result == {}
 
-    def test_contains_docker_error(self):
-        assert qemu_exec.contains_docker_error('§#§stderr§#§Unknown syscall 4001 qemu: Unsupported syscall: 4001§#§\n') is True
-        assert qemu_exec.contains_docker_error('') is False
-        assert qemu_exec.contains_docker_error(self.docker_test_output) is False
+def test_get_docker_output__wrong_arch():
+    result = qemu_exec.get_docker_output('i386', '/test_mips_static', TEST_DATA_DIR)
+    assert all(
+        b'Invalid ELF image' in b64decode(result_dict['stderr'])
+        for result_dict in result.values()
+    )
 
-    def test_process_qemu_job(self):
-        tmp = qemu_exec.test_qemu_executability
-        qemu_exec.test_qemu_executability = lambda file_path, arch_suffix, root_path: {'--option': {'stdout': 'test', 'stderr': '', 'return_code': '0'}}
 
+@pytest.mark.usefixtures('execute_docker_error')
```

Comment:
Is that nicer than just putting it as an argument? Stylewise?

Reply:
fixed (except for tests inheriting from `AnalysisPluginTest`